UKHO.MaritimeSafetyInformation.Web/UKHO.MaritimeSafetyInformation.Web/Services/NMDataService.cs

```diff
@@ -49,55 +50,57 @@ public async Task<List<ShowFilesResponseModel>> GetWeeklyBatchFiles(int year, in
                 throw;
             }
             return ListshowFilesResponseModels;
-
         }
 
-        public List<KeyValuePair<string, string>> GetAllYears(string correlationId)
+        public async Task<List<YearWeekModel>> GetAllYearWeek(string correlationId)
         {
-            List<KeyValuePair<string, string>> years = new();
-
-            _logger.LogInformation(EventIds.GetAllYearsStarted.ToEventId(), "Maritime safety information request to get all years to populate year dropdown started", correlationId);
-
-            years.Add(new KeyValuePair<string, string>("Year", ""));
-            for (int i = 0; i < 3; i++)
+            List<YearWeekModel> yearWeekModelList = new();
+            try
             {
-                string year = (DateTime.Now.Year - i).ToString();
-                years.Add(new KeyValuePair<string, string>(year, year));
-            }
-
-            return years;
-        }
-
-        public List<KeyValuePair<string, string>> GetAllWeeksOfYear(int year, string correlationId)
-        {
-            List<KeyValuePair<string, string>> weeks = new();
+                string accessToken = await _authFssTokenProvider.GenerateADAccessToken(correlationId);
 
-            _logger.LogInformation(EventIds.GetAllWeeksOfYearStarted.ToEventId(), "Maritime safety information request to get all weeks of year to populate week dropdown started", correlationId);
+                _logger.LogInformation(EventIds.GetSearchAttributeRequestDataStarted.ToEventId(), "Request to search attribute year and week data from File Share Service started for _X-Correlation-ID:{correlationId}", correlationId);
 
-            weeks.Add(new KeyValuePair<string, string>("Week Number", ""));
+                IResult<BatchAttributesSearchResponse> searchAttributes = await _fileShareService.FssSearchAttributeAsync(accessToken, correlationId);
 
-            DateTimeFormatInfo dateTimeFormatInfo = DateTimeFormatInfo.CurrentInfo;
-            DateTime lastdate;
-            if (DateTime.Now.Year == year)
-            {
-                lastdate = new DateTime(year, DateTime.Now.Month, DateTime.Now.Day);
-            }
-            else
-            {
-                lastdate = new DateTime(year, 12, 31);
+                if (searchAttributes.Data != null)
+                {
+                    for (int i = 0; i < searchAttributes.Data.BatchAttributes.Count; i++)
```

Comment:
Can we change this as a `foreach` please:
```suggestion
                    foreach (var attribute in searchAttributes.Data.BatchAttributes)
```

Reply:
implemeted foreach loop